hash.go

```diff
@@ -408,6 +396,23 @@ func (h *md5Marshal) UnmarshalBinary(b []byte) error {
 	return nil
 }
 
+func (h *md5Marshal) AppendBinary(buf []byte) ([]byte, error) {
+	d := (*md5State)(h.hashState())
+	if d == nil {
+		return nil, errors.New("crypto/md5: can't retrieve hash state")
+	}
+
+	buf = append(buf, md5Magic...)
+	buf = appendUint32(buf, d.h[0])
+	buf = appendUint32(buf, d.h[1])
+	buf = appendUint32(buf, d.h[2])
+	buf = appendUint32(buf, d.h[3])
+	buf = append(buf, d.x[:d.nx]...)
+	buf = append(buf, make([]byte, len(d.x)-int(d.nx))...) // already zero
```

Comment:
I don't think the comment is correct in the new context, at least the way I read it. This code isn't always dealing with preallocated zeros--the zeros may or may not already exist.

I don't think this needs any comment anymore. The new approach is more intuitive, "normal" Go: it's clear from the line alone that we're appending bytes of 0, whether or not they already existed.

(Same for the other `// already zero` comments being moved.)